packages/neuron-wallet/src/services/wallets.ts

```diff
@@ -293,64 +285,85 @@ export default class WalletService {
+      capacity: (BigInt(item.capacity) * BigInt(1)).toString(),
     }))
 
-    const rawTransaction = await TransactionsService.generateTx(lockHashes, targetOutputs, changeAddress)
+    const changeAddress: string = this.getChangeAddress()
 
-    const txHash = await (core.rpc as any).computeTransactionHash(rawTransaction)
-    const signature = addrObj.sign(txHash)
-    const signatureSize = core.utils.hexToBytes(signature).length
-    const sequence = new DataView(new ArrayBuffer(8))
-    sequence.setUint8(0, signatureSize)
-    const sequencedSignatureSize = Buffer.from(sequence.buffer).toString('hex')
-    const witness = {
-      data: [`0x${addrObj.publicKey}`, `0x${signature}`, `0x${sequencedSignatureSize}`],
-    }
-    const witnesses = Array.from(
-      {
-        length: rawTransaction.inputs.length,
-      },
-      () => witness
-    )
-    rawTransaction.witnesses = witnesses
-    const realTxHash = await core.rpc.sendTransaction(rawTransaction)
+    const tx: TransactionWithoutHash = await TransactionsService.generateTx(lockHashes, targetOutputs, changeAddress)
+
+    const txHash: string = await (core.rpc as any).computeTransactionHash(ConvertTo.toSdkTxWithoutHash(tx))
+
+    const { inputs } = tx
+
+    const witnesses: Witness[] = inputs!.map((input: Input) => {
+      const blake160: string = input.lock!.args![0]
+      const info = addressInfos.find(i => i.blake160 === blake160)
+      const { path } = info!
+      const privateKey = this.getPrivateKey(path)
+      const witness = this.signWitness({ data: [] }, privateKey, txHash)
+      return witness
+    })
+
+    tx.witnesses = witnesses
+
+    const txToSend = ConvertTo.toSdkTxWithoutHash(tx)
+    await core.rpc.sendTransaction(txToSend)
 
     TransactionsService.txSentSubject.next({
-      transaction: rawTransaction,
-      txHash: realTxHash,
+      transaction: tx,
+      txHash,
     })
 
     return txHash
   }
+
+  public getAddressInfo = () => {
+    const item = {
+      pubkey: '0x024a501efd328e062c8675f2365970728c859c592beeefd6be8ead3d901330bc01',
+      address: 'ckt1q9gry5zgxmpjnmtrp4kww5r39frh2sm89tdt2l6v234ygf',
+      blake160: '0x36c329ed630d6ce750712a477543672adab57f4c',
+      path: '',
+    }
+    return [item]
+  }
+
+  public getChangeAddress = (): string => {
+    return 'ckt1q9gry5zgxmpjnmtrp4kww5r39frh2sm89tdt2l6v234ygf'
+  }
+
+  public signWitness = (witness: Witness, privateKey: string, txHash: string): Witness => {
+    const addrObj = core.generateAddress(privateKey)
+    const oldData = witness.data
+    const blake2b = new Blake2b()
+    blake2b.update(txHash)
+    oldData.forEach(data => {
+      blake2b.update(data)
+    })
+    const message = blake2b.digest()
+    const signature = `0x${addrObj.sign(message)}`
+    const newWitness: Witness = {
+      data: [`0x${addrObj.publicKey}`, signature],
+    }
+    return newWitness
+  }
+
+  public getPrivateKey = (path: string): string => {
+    if (path !== '') {
```

Comment:
Should path always be `` 🤯?

Reply:
It's a mocked path and updated to use real path format.